### PR TITLE
fix: print the correct violation message

### PR DESCRIPTION
**Pull request by jfroche**
we don't want to rely on the boolean value of the xml element but on the
boolean value of the message contained in that element

### CODE REVIEW

File: run_codenarc.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def _print_violations(package_file_path, violations):
     """
     for violation in violations:
         message_element = violation.find('Message')
-        message = message_element.text if message_element else '[empty message]'
+        message = message_element.text or '[empty message]'
```
Comment (ala-ableton) [inline run_codenarc.py]: `violation.find('Message')` returns `None` if the `violation` element doesn't have a `Message` sub-element. If `message_element` is `None`, `message_element.text` will raise an `AttributeError`. See https://github.com/AbletonDevTools/groovylint/commit/88ca1aa4b94797f3fd9858a6943d3072851373f0 for more details.

**Comment by jfroche** [comment]
```
# python3 /opt/run_codenarc.py -- -includes='./Jenkinsfile' -rulesetfiles=file:/ws/ruleset.groovy
INFO org.codenarc.ruleregistry.PropertiesFileRuleRegistry - Loaded properties file in 44ms; 383 rules
INFO org.codenarc.ruleset.GroovyDslRuleSet - Loading ruleset from [file:/ws/ruleset.groovy]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/basic.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/braces.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/comments.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/convention.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/design.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/exceptions.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/formatting.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/generic.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/groovyism.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/imports.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/naming.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/security.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/size.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/unnecessary.xml]
INFO org.codenarc.ruleset.XmlFileRuleSet - Loading ruleset from [rulesets/unused.xml]
INFO org.codenarc.ruleset.PropertiesFileRuleSetConfigurer - RuleSet configuration properties file [codenarc.properties] not found.
INFO org.codenarc.report.AbstractReportWriter - Report file [codenarc-report.xml] created.
INFO Scanned 1 files
ERROR ./Jenkinsfile:3: TrailingComma: [empty message]
ERROR ./Jenkinsfile:6: LineLength: [empty message]
ERROR ./Jenkinsfile:6: UnnecessaryGString: [empty message]
ERROR ./Jenkinsfile:7: LineLength: [empty message]
ERROR ./Jenkinsfile:7: UnnecessaryGString: [empty message]
ERROR ./Jenkinsfile:11: SpaceBeforeOpeningBrace: [empty message]
ERROR ./Jenkinsfile:13: Indentation: [empty message]
ERROR ./Jenkinsfile:14: Indentation: [empty message]
ERROR ./Jenkinsfile:15: Indentation: [empty message]
ERROR ./Jenkinsfile:16: Indentation: [empty message]
ERROR ./Jenkinsfile:18: Indentation: [empty message]
ERROR ./Jenkinsfile:19: Indentation: [empty message]
ERROR ./Jenkinsfile:20: Indentation: [empty message]
ERROR ./Jenkinsfile:21: Indentation: [empty message]
ERROR ./Jenkinsfile:24: Indentation: [empty message]
ERROR ./Jenkinsfile:25: Indentation: [empty message]
ERROR ./Jenkinsfile:29: Indentation: [empty message]
ERROR ./Jenkinsfile:30: Indentation: [empty message]
ERROR ./Jenkinsfile:31: Indentation: [empty message]
ERROR ./Jenkinsfile:40: Indentation: [empty message]
ERROR ./Jenkinsfile:40: LineLength: [empty message]
ERROR ./Jenkinsfile:42: BlockEndsWithBlankLine: [empty message]
ERROR Found 22 violation(s)
```
if we add a pdb:
```
> /opt/run_codenarc.py(109)_print_violations()
-> message = message_element.text if message_element else '[empty message]'
(Pdb) bool(message_element)
False
(Pdb) message_element.text
'List should contain trailing comma.'
(Pdb) bool(message_element.text)
True
(Pdb) message_element
<Element 'Message' at 0x7f44d314d458>o
```

The boolean evaluation code for an Element is here: https://github.com/python/cpython/blob/master/Lib/xml/etree/ElementTree.py#L214
`message_element` has no children so it is False

**Comment by ala-ableton** [comment]
Thanks for the detailed report, then we should change this line to
```python
        message = message_element.text if message_element`is not None else '[empty message]'
``

**Comment by jfroche** [comment]
ok for me

**Comment by nre-ableton** [comment]
@jfroche I'm closing this PR in favor of https://github.com/AbletonDevTools/groovylint/pull/107. The reason being is that our CI system can't yet build external pull requests (hence the failing commit status check on this one). Thanks for the fix, and let's continue the discussion over there!

**Pull request closed without merging**